src/base/math.cc

```diff
@@ -17,4 +17,35 @@
  *
  */
 
-// Dummy file as cmake cannot handle header only libraries :(.
+#include "base/math.h"
+
+#include <regex>
+
+#include "base/wexception.h"
+
+namespace math {
+
+// This function has RST documentation in logic/map_objects/map_object_program
+unsigned read_percent_to_int(const std::string& input) {
+	std::smatch match;
+	std::regex re("^(\\d+)([.](\\d{1,2})){0,1}%$");
+	if (std::regex_search(input, match, re)) {
+		// Convert to range
+		uint64_t result =
+		   100U * std::stoul(match[1]) +
+		   (match[3].str().empty() ?
```

Comment:
Checking for match[1] / match[3] instead of match[0] / match[2] looks a bit strange. If you know why that is so, can you please add two short comments about matches 0 and 2, which a reader of the code might expect?